src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -191,6 +191,22 @@ private Type EmitCoercionCode(AssociativeNode arg, Type argType, ParameterInfo p
 
         private Type EmitIListCoercion<T>(AssociativeNode arg)
         {
+            LocalBuilder localBuilder;
+            // Load array to be coerced.
+            int currentVarIndex = localVarIndex;
+            if (arg is IdentifierNode ident)
+            {
+                currentVarIndex = variables[ident.Value].Item1;
+            }
+            else
+            {
+                localBuilder = DeclareLocal(typeof(IList), "IList to coerce");
+                currentVarIndex = localBuilder.LocalIndex;
+
+                EmitOpCode(OpCodes.Stloc, currentVarIndex);
+                EmitOpCode(OpCodes.Ldloc, currentVarIndex);
```

Comment:
do we really need to do a pop(stloc) then push(ldloc) ? seems redundant...but maybe that is how it's done?

Reply:
Yes, we do, since we need to store the value at the top of the stack in the local var at `currentVarIndex`, so it can be loaded from that index later when we need it.